src/view/com/util/List.web.tsx

```diff
@@ -242,12 +250,47 @@ let Row = function RowImpl<ItemT>({
     | undefined
     | ((data: {index: number; item: any; separators: any}) => React.ReactNode)
   extraData: any
+  onItemSeen?: (item: ItemT) => void
 }): React.ReactNode {
+  const tailRef = React.useRef(null)
+  const isIntersecting = React.useRef(false)
+
+  const handleIntersection = useNonReactiveCallback(
+    (entries: IntersectionObserverEntry[]) => {
+      batchedUpdates(() => {
+        if (!onItemSeen) {
+          return
+        }
+        entries.forEach(entry => {
+          if (entry.isIntersecting !== isIntersecting.current) {
+            isIntersecting.current = entry.isIntersecting
+            if (entry.isIntersecting) {
+              onItemSeen!(item)
+            }
+          }
+        })
+      })
+    },
+  )
+
+  React.useEffect(() => {
+    if (!onItemSeen) {
+      return
+    }
+    const observer = new IntersectionObserver(handleIntersection)
+    const tail: Element | null = tailRef.current!
```

Comment:
nit: I only called this `tail` because it was literally tracking the tail of the list, this is more like `row` and `rowRef`.